libcudacxx/include/cuda/std/__execution/env.h

```diff
@@ -0,0 +1,319 @@
+ * sub-environments.
+ *
+ * @struct get_env_t
+ * A callable object for retrieving the environment associated with an object.
+ */
+
+#define _CCCL_API         _CCCL_HOST_DEVICE _CCCL_VISIBILITY_HIDDEN _CCCL_EXCLUDE_FROM_EXPLICIT_INSTANTIATION
+#define _CCCL_TRIVIAL_API _CCCL_API _CCCL_FORCEINLINE _CCCL_ARTIFICIAL _CCCL_NODEBUG
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+namespace detail
+{
+template <class _Ty, class _Query>
+auto __query_result_() -> decltype(declval<_Ty>().query(_Query()));
+
+#if _CCCL_HAS_EXCEPTIONS()
+template <class _Ty, class _Query>
+using __nothrow_queryable_with_ _CCCL_NODEBUG_ALIAS = enable_if_t<noexcept(declval<_Ty>().query(_Query{}))>;
+#endif
+
+template <class _Ty>
+extern _Ty __unwrap_ref;
+
+template <class _Ty>
+extern _Ty& __unwrap_ref<::std::reference_wrapper<_Ty>>;
+
+template <class _Ty>
+extern _Ty& __unwrap_ref<reference_wrapper<_Ty>>;
+
+inline constexpr size_t __npos = static_cast<size_t>(-1);
+
+_CCCL_API constexpr auto __find_pos(bool const* const __begin, bool const* const __end) noexcept -> size_t
+{
+  for (bool const* __where = __begin; __where != __end; ++__where)
+  {
+    if (*__where)
+    {
+      return static_cast<size_t>(__where - __begin);
+    }
+  }
+  return __npos;
+}
+} // namespace detail
+
+namespace execution
+{
+template <class _Ty, class _Query>
+using __query_result_t _CCCL_NODEBUG_ALIAS = decltype(detail::__query_result_<_Ty, _Query>());
+
+template <class _Ty, class _Query>
+_CCCL_CONCEPT __queryable_with = _IsValidExpansion<__query_result_t, _Ty, _Query>::value;
+
+#if _CCCL_HAS_EXCEPTIONS()
+template <class _Ty, class _Query>
+_CCCL_CONCEPT __nothrow_queryable_with = _IsValidExpansion<detail::__nothrow_queryable_with_, _Ty, _Query>::value;
+#else
+template <class _Ty, class _Query>
+_CCCL_CONCEPT __nothrow_queryable_with = true;
+#endif
+
+template <class _Ty>
+using __unwrap_reference_t _CCCL_NODEBUG_ALIAS = decltype(detail::__unwrap_ref<_Ty>);
+
+/**
+ * @brief A template structure representing a query with a query and a value.
+ *
+ * @tparam _Query The type of the query associated with the query.
+ * @tparam _Value The type of the value associated with the query.
+ *
+ * This structure provides a mechanism to associate a query with a value
+ * and allows querying the value using the query type.
+ *
+ * Members:
+ * - _Query __query: The query object.
+ * - _Value __value: The value associated with the query.
+ *
+ * Member Functions:
+ * - constexpr auto query(_Query) const noexcept -> const _Value&:
+ *   Returns the value associated with the query. This function is marked
+ *   as `constexpr` and `noexcept`, ensuring it can be evaluated at compile-time
+ *   and does not throw exceptions.
+ */
+template <class _Query, class _Value>
+struct _CCCL_TYPE_VISIBILITY_DEFAULT prop
+{
+  _CCCL_NO_UNIQUE_ADDRESS _Query __query;
+  _CCCL_NO_UNIQUE_ADDRESS _Value __value;
+
+  _CCCL_TRIVIAL_API constexpr auto query(_Query) const noexcept -> const _Value&
```

Comment:
Missing `[[nodiscard]]`, please audit
```suggestion
  [[nodiscard]] _CCCL_TRIVIAL_API constexpr auto query(_Query) const noexcept -> const _Value&
```